### PR TITLE
Add network functions on the DPU side

### PR DESCRIPTION
**Pull request by bn222**

### CODE REVIEW

File: daemon/sfc-reconciler/sfc.go [hunk 1]
```diff
@@ -26,20 +27,31 @@ type SfcReconciler struct {
...
+			Namespace: "dpu-operator-system",
+			Annotations: map[string]string{
+				"k8s.v1.cni.cncf.io/networks": "dpunfcni-conf, dpunfcni-conf",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  name,
+					Image: image,
```
Comment (wizhaoredhat) [inline daemon/sfc-reconciler/sfc.go:43]: We might want to have a PullPolicy = Always for debugging here.

File: daemon/dpudaemon.go [hunk 2]
```diff
@@ -128,6 +128,15 @@ func (d *DpuDaemon) ListenAndServe() error {
 	    }
 	}()
 
+	go func() {
+		d.log.Info("Starging CNI server")
```
Comment (wizhaoredhat) [inline daemon/dpudaemon.go:151]: Indentation and SP "Starting". I think you need to run "go fmt".

File: api/v1/classifierconfig_types.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ClassifierConfigSpec defines the desired state of ClassifierConfig
+type ClassifierConfigSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of ClassifierConfig. Edit classifierconfig_types.go to remove/update
+	Foo string `json:"foo,omitempty"`
```
Comment (wizhaoredhat) [inline api/v1/classifierconfig_types.go:32]: Are we missing a real input for this?

File: daemon/dpudaemon.go [hunk 4]
```diff
@@ -106,11 +112,34 @@ func (d *DpuDaemon) Listen() (net.Listener, error) {
...
-	lis, err := d.Listen()
+	done := make(chan error, 1)
+	listener, err := d.Listen()
+
 	if err != nil {
+		d.log.Error(err, "Failed to listen")
 		return err
 	}
-	return d.Serve(lis)
+	go func() {
```
Comment (wizhaoredhat) [inline daemon/dpudaemon.go:141]: This implementation does not work:
```package main

import (
	"errors"
	"fmt"
	"time"
)

func something(c chan error) {
	fmt.Println("HELLO1")
	c <- nil
}

func something3(c chan error) {
	fmt.Println("HELLO1")
	time.Sleep(5 * time.Second)
	c <- errors.New("an error")
}

func something2() error {
	c := make(chan error, 1)
	go something(c)
	go something3(c)
	return <-c
}
func main() {
	err := something2()
	fmt.Println(err)
}
```
It does not wait for all the go routines to finish, you need a wait group.

Comment (wizhaoredhat) [reply daemon/dpudaemon.go:141]: Something like this works:
```
package main

import (
	"errors"
	"fmt"
	"sync"
	"time"
)

func test1(wg *sync.WaitGroup, c chan error) {
	defer wg.Done()
	fmt.Println("HELLO1")
	c <- nil
}

func test2(wg *sync.WaitGroup, c chan error) {
	defer wg.Done()
	fmt.Println("HELLO1")
	time.Sleep(5 * time.Second)
	c <- errors.New("an error")
}

func testing() chan error {
	c := make(chan error, 2)
	var wga sync.WaitGroup
	wga.Add(1)
	go test1(&wga, c)
	wga.Add(1)
	go test2(&wga, c)
	wga.Wait()
	close(c)
	return c
}
func main() {
	errors := testing()
	for err := range errors {
		fmt.Println(err)
	}
}
```

File: daemon/hostdaemon.go [hunk 5]
```diff
@@ -180,13 +194,33 @@ func (d *HostDaemon) Listen() (net.Listener, error) {
...
 func (d *HostDaemon) ListenAndServe() error {
+	done := make(chan error, 1)
 	listener, err := d.Listen()
 
 	if err != nil {
 		d.log.Error(err, "Failed to listen")
 		return err
 	}
-	return d.Serve(listener)
+	go func() {
```
Comment (wizhaoredhat) [inline daemon/hostdaemon.go:204]: Same comment as above.

**Review by wizhaoredhat — CHANGES_REQUESTED** [review]

File: daemon/hostdaemon.go (hunk 5 above)
Comment (wizhaoredhat) [reply daemon/hostdaemon.go:203]: Missing Workgroup?

Comment (bn222) [reply daemon/hostdaemon.go:203]: yes, thanks.

File: daemon/main.go [hunk 6]
```diff
@@ -80,12 +81,21 @@ func makeExecutable(file string) error {
 	return nil
 }
 
+func writable(path string) bool {
+	return unix.Access(path, unix.W_OK) == nil
```
Comment (wizhaoredhat) [inline daemon/main.go:85]: This does not work. I tried it before.

Comment (wizhaoredhat) [reply daemon/main.go:85]: I think I removed it priorly.

**Pull request merged**